templates/single-quiz/pagination.php

```diff
@@ -44,12 +44,16 @@
 				]
 			)
 		);
+
+		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- No need to escape the pagination.
+		echo Sensei()->quiz->replace_pagination_links_with_buttons( $sensei_pagination_list );
```

Comment:
Replacing all the pagination links with buttons in order to submit the form and save the state on each page change. Also, the already existing state saving mechanism is utilized by using this method. All other options require JS, which could lead to a more complicated solution.